### PR TITLE
support larger than 1k client responses (bigger winfmts, etc)

### LINKED ISSUE #39: large window lists are truncated
it seems that a larger output for "windows" than 1k will be truncated:
```
$ rp () { sdorfehs -c "$*"; }

$ rp windows | wc -c # this looks fine
816

$ rp windows %{i,a,c,n} | wc -c # truncated
1022

$ rp windows %{i,a,c,n,t} | wc -c # same, yet should be bigger
1022

$ rp windows `perl -e "print 'a' x 1000"` | wc -c # line for each window!
1022
```
this does not happen with ratpoison btw

### PR DESCRIPTION
**Pull request by smemsh**
Previously the buffer size could not exceed 1k in the response from server, or it would be truncated.  This included commands like, for example:

```
$ windows w %{i,a,c,t,n}
```

would easily get over 1024 chars (actually 1022 due to \n and \0) on my system.  The new code can handle any size response.

Fixes #39

### CODE REVIEW

**Comment by smemsh** [comment]
Too bad about all the `#ifdef DEBUG`, but `PRINT_DEBUG()` isn't really useful in the client (it prints to stdout) and `warnx()` could not be used without condition because those print to stderr, which non-debug case should not have to deal with.

I debated removing these prints after the code was working, but it took some time to develop the right points where those debug prints should be located, I had an occasional off-by-one that was difficult to track down.  A new macro like `PRINT_DEBUG()` but printing to stderr would be better for these, or even a client flag for cli to turn this on and off, but then we would need args processing.

For now, thought it best to ask for comment on what should be done with these prints...

**Comment by smemsh** [comment]
> Too bad about all the #ifdef DEBUG, but PRINT_DEBUG() isn't really useful in the client (it prints to stdout)

I added a different macro `WARNX_DEBUG()` that uses `stderr`, it's only used in the one function now, but it reads a lot better while retaining the important print info for debugging, at least for me.  This function is special anyways because it's for cli and used in scripts, it has to use `stderr` so as to be out-of-band.

This will not print anything if `-DDEBUG` is not defined.

Let me know if it's suitable, maybe you won't like those prints even in your debug builds.

**Comment by smemsh** [comment]
now I further hid those stderr macros under `-DSENDCMD_DEBUG` instead of just `-DDEBUG` because it's probably only useful in subsystem-specific scenario just like `-DINPUT_DEBUG`

File: communications.c [hunk 1]
```diff
@@ -82,10 +85,20 @@ int
...
 	size_t len;
-	int fd;
+	ssize_t count;
+	int fd, firstloop;
+	int flags = 0x0;
+	FILE *outf = NULL;
+
+#ifdef SENDCMD_DEBUG
+	pid_t pid = getpid();
+	char *dpfx = xsprintf("send_command_%d", pid);
```
Comment (jcs) [inline communications.c:99]: You are leaking `dpfx` here, there is no `free(dpfx)`.

**Review by jcs — CHANGES_REQUESTED** [review]

File: communications.c (hunk 1 above)
Comment (smemsh) [reply communications.c:99]: good catch, thanks!

**Pull request merged**